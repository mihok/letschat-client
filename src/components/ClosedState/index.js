import { h } from 'preact';
import PropTypes from 'prop-types';

import Header from '../Header';

import './styles.css';

const ClosedState = props => (
  <Header chatOpen={props.chatOpen} toggleChat={() => props.toggleChat(true)} />
);

ClosedState.defaultProps = {
  chatOpen: false,
};

ClosedState.propTypes = {
  toggleChat: PropTypes.func.isRequired,
  chatOpen: PropTypes.bool,
};

export default ClosedState;
