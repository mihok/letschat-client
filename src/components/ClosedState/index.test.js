import { h } from 'preact';
import render from 'preact-render-to-string';

import ClosedState from '.';

describe('<ClosedState />', () => {
  it('matches snapshot', () => {
    const tree = render(<ClosedState toggleChat={jest.fn()} />);
    expect(tree).toMatchSnapshot();
  });
});
