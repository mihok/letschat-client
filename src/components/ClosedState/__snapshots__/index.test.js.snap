// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClosedState /> matches snapshot 1`] = `"<header style=\\"background-color: test;\\" role=\\"button\\" tabIndex=\\"0\\" class=\\"Header--messenger\\"><span class=\\"Header__title--messenger\\">Chat with John</span><button class=\\"Header__closeBtn--messenger\\"><span>...</span></button></header>"`;
